From 494fbc5e92af3f60eb213c27d9d07d8630744177 Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Wed, 5 Nov 2014 16:05:00 +0900
Subject: [PATCH 17/22] Revert "workaround: refer to the global gl_renderer
 when destroying surfaces."

This reverts commit ec0ef86f2024622765e3d0ee76f285f6842ea55f.
---
 src/gl-renderer.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/gl-renderer.c b/src/gl-renderer.c
index 507bd2b..63af75d 100644
--- a/src/gl-renderer.c
+++ b/src/gl-renderer.c
@@ -1321,17 +1321,18 @@ surface_state_destroy(struct gl_surface_state *gs, struct gl_renderer *gr)
 	free(gs);
 }
 
-static struct gl_renderer *__gr;
-
 static void
 surface_state_handle_surface_destroy(struct wl_listener *listener, void *data)
 {
 	struct gl_surface_state *gs;
+	struct gl_renderer *gr;
 
 	gs = container_of(listener, struct gl_surface_state,
 			  surface_destroy_listener);
 
-	surface_state_destroy(gs, __gr);
+	gr = get_renderer(gs->surface->compositor);
+
+	surface_state_destroy(gs, gr);
 }
 
 static void
@@ -1952,8 +1953,6 @@ gl_renderer_create(struct weston_compositor *ec, EGLNativeDisplayType display,
 	ec->capabilities |= WESTON_CAP_ROTATION_ANY;
 	ec->capabilities |= WESTON_CAP_CAPTURE_YFLIP;
 
-	__gr = gr;
-
 	if (gl_renderer_setup_egl_extensions(ec) < 0)
 		goto err_egl;
 
-- 
1.7.9.5

